app/controllers/applications/process_controller.rb

```diff
@@ -186,6 +186,8 @@ def benefit_check_and_redirect(benefits)
       if benefits
         benefit_check_runner.run
         determine_override
+      elsif no_benefits_paper_evidence?
```

Comment:
I see you have created an ApplicationFormSave class to tidy up some of the controller (looks more like a repository to me), but then we have this business logic in the controller that has been added to.  Could this not do with a dose of refactoring (not necessarily in this PR) to get the controller being super slim and just providing the http API to the underlying services ?